### ISSUE TITLE
ReduceLROnPlateau will fail when add new parameter group to the optimizer

### ISSUE THREAD

**Issue by thuwyh** [pytorch/pytorch#20997]
## 🐛 Bug

ReduceLROnPlateau will fail when add new parameter group to the optimizer.
The `_reduce_lr` function will raise list index out of range error

## To Reproduce

1. initialize optimizer: `optimizer = Adam(filter(lambda p: p.requires_grad, model.parameters()), args.lr)`
1. initialize scheduler: `scheduler = ReduceLROnPlateau(optimizer, patience=1, factor=0.1, verbose=True, mode='max')`
1. add parameter group to the optimizer: `optimizer.add_param_group({'params':unfreezed_params, 'lr':lr})`
1. raise error when the learning rate should be changed

## Expected behavior

when add a parameter group to the optimizer, the `min_lrs` attribute of the scheduler should be updated to avoid this error.

## Environment

 - PyTorch Version (1.1.0):

**Comment by icedpanda** [comment]
I'm having the same issue as well.

## Environment
- Pytorch version: 1.8.1
- OS windows

**Comment by upayuryeva** [comment]
> I'm having the same issue as well.
> 
> ## Environment
> * Pytorch version: 1.8.1
> * OS: Windows
Me too :(
Did you solve this?

**Comment by patrontheo** [comment]
@ailzhang  @ezyang @malfet 
Same issue, here, any plans to solve this ?

**Comment by patrontheo** [comment]
Fixed by #137637. 
@janeyx99 I think this one can be closed too.
(@florianmf)

**Comment by janeyx99** [comment]
yep!

**Comment by kikatuso** [comment]
```
def update_optimizer(epoch,optimizer,model,scheduler=None,new_lr=1e-4):

    print('Updating the optimiser...')
    new_params = [param for name, param in model.named_parameters() if any(layer in name for layer in model.unfreeze_scheduler[epoch]) and param.requires_grad]
    optimizer.add_param_group({'params': new_params, 'lr': new_lr})
    if scheduler is not None:
        scheduler.min_lrs = [0] * len(optimizer.param_groups)
    
```

Still not resolved, however, a quick workaround makes it work.

**Comment by janeyx99** [comment]
@kikatuso Wait what error do you run into? I'd expect the above code to work on main now.
